packages/react-badge/src/components/PresenceBadge/usePresenceBadge.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { makeMergeProps } from '@fluentui/react-utilities';
+import { PresenceBadgeProps, PresenceBadgeState, PresenceBadgeStatus } from './PresenceBadge.types';
+import { useBadge, BadgeProps } from '../Badge/index';
+import {
+  SkypeMinusIcon,
+  SkypeClockIcon,
+  SkypeArrowIcon,
+  SkypeCheckIcon,
+  CancelIcon,
+} from './DefaultPresenceBadgeIcons';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const presenceBadgeShorthandProps: (keyof PresenceBadgeProps)[] = ['icon'];
+
+const mergeProps = makeMergeProps<PresenceBadgeState>({ deepMerge: presenceBadgeShorthandProps });
+
+const iconMap: Record<PresenceBadgeStatus, JSX.Element> = {
+  busy: <SkypeMinusIcon />,
+  available: <SkypeCheckIcon />,
+  away: <SkypeClockIcon />,
+  offline: <CancelIcon />,
+  outOfOffice: <SkypeArrowIcon />,
+  doNotDisturb: <></>,
+};
+
+/**
+ * Returns the props and state required to render the component
+ */
+export const usePresenceBadge = (
+  props: PresenceBadgeProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: PresenceBadgeProps,
+): PresenceBadgeState => {
+  const state = useBadge(
+    props,
+    ref,
+    mergeProps(
+      {
+        size: 'small',
+        status: 'available',
+        outOfOffice: false,
+      },
+      defaultProps,
+    ) as BadgeProps,
+  ) as PresenceBadgeState;
+
+  if (!state.icon) {
+    state.icon = { children: () => iconMap[state.status] };
+  }
```

Comment:
It's possible that someone would add props to the icon slot but expect that the Badge would add the icon itself. So you should check `state.icon.children` in addition to `state.icon`. 

Or better, simplify it by making sure that `state.icon` is never undefined, by providing a default in `mergeProps`:
`icon: { as: 'div' },` (or `'span'`?). And then you only need to check if `state.icon.children` is undefined. I'll add a suggestion above to make it clear what I mean there.

Also out of curiosity, why use a children render function instead of simply using the icon directly as children?

```suggestion
  if (!state.icon.children) {
    state.icon.children = iconMap[state.status];
  }
```